### DISCUSSION TITLE
IPAdapter_face

### DISCUSSION THREAD

**Discussion by barepixels** [MoonRide303/Fooocus-MRE#164] [Ideas]
Recently lllyasviel added **IPAdapter** to his Fooocus and called it **Image Prompts (Midjourney Image Prompts)**

However, lllyasviel did not add **IPAdapter plus face** because it's only available for SD1.5

So I looked into what can **IPAdapter plus face** do... OMG it's AMAZING.  Let me show you

input image: (prompt: closeup photo of a guy flexing muscles at the gym)
![einstein](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/aaa584b5-7425-4b07-8312-82066661ce69)

Output:
![01](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/4eb61e19-e8f4-4c28-aa93-13dd9b7cef8a)
![02](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/8232bd83-af50-4e62-b19e-ec6d8ff8dbfb)
![03](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/f3f30fc6-d072-46a7-bc6b-fb1cce44b94e)
![04](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/932eded5-a052-47cb-ad89-7033602e2559)

It's very fast.  MO MASK, NO INPAINT, NO LORA needed  THIS IS A GAME CHANGER

@MoonRide303, since Fooocus-MRE supports SD1.5, can you look into implementing **IPAdapter_face**   Maybe you can look at llyasviel's codes or ask llyasviel for assistance.  IMHO **IPAdapter_face** is more useful than other IPAdapter

Here is the official link with ComfyUI json file(s):
https://github.com/cubiq/ComfyUI_IPAdapter_plus

**Comment by vicento** [comment]
**great !** 
_**how to use it on Fooocus-MRE or Fooocus ?**_

**Reply by barepixels** [reply]
you can't yet.  author(s) have to add the feature

**Comment by barepixels** [comment]
IPAdapter plus face for SDXL is available now  https://github.com/cubiq/ComfyUI_IPAdapter_plus

Tested  it works YAY.  Model used https://civitai.com/models/139562?modelVersionId=169921
![woman](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/e56ef2ab-bc18-4761-bed4-e4ba4d388d27)

Generated
![IPAdapter_00057_](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/dff4008c-a1a2-467e-a878-3cdcba83d8dc)
![IPAdapter_00042_](https://github.com/MoonRide303/Fooocus-MRE/assets/92839033/3f048039-60f6-4644-875e-db1fed2f2c16)

**Comment by naiyansui2023** [comment]
![微信图片_20231107205610](https://github.com/MoonRide303/Fooocus-MRE/assets/130981007/c836a260-af32-478e-ab36-93ec04cf93c9)

**Reply by TanvirHafiz** [reply]
that does not really look like einstein, original fooocus usually does a better job!
